### ISSUE TITLE
DPE Inspector: Modifying most of the Sky Atmosphere component's properties causes the Editor to crash

### ISSUE THREAD

**Issue by LB-ArturZieba** [o3de/o3de#16723]
**Describe the bug**
If `ed_enableDPE` is `true`, then modifying most of the *Sky Atmosphere* component's properties (for example *Atmosphere height* or *Ozone absorption scale*) causes the Editor to crash.

Call Stack:
```
Call Stack Trace:
25) 00007FF8506CDEA7 (KERNELBASE) : UnhandledExceptionFilter
24) 00007FF84FFCC888 (InProcessClient64) : (function-name not available)
23) 00007FF84FFCC4D1 (InProcessClient64) : (function-name not available)
22) 00007FF852DB5330 (ntdll) : memset
21) 00007FF852D9C876 (ntdll) : _C_specific_handler
20) 00007FF852DB221F (ntdll) : _chkstk
19) 00007FF852D614B4 (ntdll) : RtlRaiseException
18) 00007FF852DB0D2E (ntdll) : KiUserExceptionDispatcher
17) 00007FF81C79D319 (Qt5Widgets) : QAbstractSpinBox::selectAll
16) 00007FF81C79E091 (Qt5Widgets) : QAbstractSpinBox::stepBy
15) 00007FF81C79EF8D (Qt5Widgets) : QAbstractSpinBox::wheelEvent
14) 00007FF81C65793E (Qt5Widgets) : QWidget::event
13) 00007FF81C6349A1 (Qt5Widgets) : QApplicationPrivate::notify_helper
12) 00007FF81C633A23 (Qt5Widgets) : QApplication::notify
11) 00007FF81A7817CB (Qt5Core) : QCoreApplication::notifyInternal2
10) C:\o3de\Code\Framework\AzQtComponents\AzQtComponents\Components\GlobalEventFilter.cpp (73) : AzQtComponents::GlobalEventFilter::eventFilter
 9) C:\o3de\Code\Editor\Core\QtEditorApplication.cpp (140) : `anonymous namespace'::EditorGlobalEventFilter::eventFilter
 8) 00007FF81A783764 (Qt5Core) : QCoreApplicationPrivate::sendThroughApplicationEventFilters
 7) 00007FF81C6348E3 (Qt5Widgets) : QApplicationPrivate::notify_helper
 6) 00007FF81C633A23 (Qt5Widgets) : QApplication::notify
 5) 00007FF81A7817CB (Qt5Core) : QCoreApplication::notifyInternal2
 4) 00007FF81B9375F5 (Qt5Gui) : QGuiApplicationPrivate::processWheelEvent
 3) 00007FF81B91EE99 (Qt5Gui) : QWindowSystemInterface::sendWindowSystemEvents
 2) 00007FF81A7CA84A (Qt5Core) : QEventDispatcherWin32::processEvents
 1) 00007FF841D13799 (qwindows) : qt_plugin_query_metadata
```

It should be noted that this does not occur if `ed_enableDPE` is set to `false`.

This issue occurs on both Windows and Linux.

**Steps to reproduce**
1. Execute `ed_enable 1` command in the *Console*.
2. Create an entity and add *Sky Atmosphere* component to it.
3. Modify different properties on the *Sky Atmosphere* component.

**Expected behavior**
Editor remains stable.

**Actual behavior**
Editor crashes.

**Video, Editor.log, error.log, and error.dmp**

https://github.com/o3de/o3de/assets/86952082/e0b550c3-eff1-4747-bc82-96cce2c51620

[Editor.log](https://github.com/o3de/o3de/files/12619782/Editor.log)
[error.log](https://github.com/o3de/o3de/files/12619783/error.log)
[error.zip](https://github.com/o3de/o3de/files/12619784/error.zip) - contains `error.dmp` file.

**Found in Branch**
Stabilization/2310 (372b2b8)

**Commit ID from [o3de/o3de](https://github.com/o3de/o3de) Repository**
372b2b829a767a8b522b05acc4b27810dbf49ad7

**Desktop**
 - Device: PC
 - OS: Windows
 - Version 10
 - CPU AMD Ryzen 5 3600X 6-Core Processor
 - GPU AMD Radeon RX 5700 XT
 - Memory 16GB

**Comment by LB-ArturZieba** [comment]
It should be noted that this was also observed on the Stabilization/2310 (13f3cbe) build on the *Light* component.

**Comment by LB-ArturZieba** [comment]
This issue was also observer on *Recast Navigation Mesh* component on the Stabilization/2310(5094225) build.

**Comment by cgalvan** [comment]
https://github.com/o3de/o3de/issues/16722

**Comment by LB-ArturZieba** [comment]
Issue verified as fixed on the Stabilization/2310 (d380ffa) branch.
